Liste sorusu ( Adreslere denk gelen hashler)

Roq4 opened the pull request.

Roq4 committed:
Liste sorusu ( Adreslere denk gelen hashler)

mnsaglam commented on liste/liste.js:
```diff
@@ -20,43 +20,43 @@ const NodeURLleri = {
+    // Her adres sorgulanır.
+    for (const adres of adresler) {
+      try {
+        const handle = await jsonrpc.call(NodeURLleri[ağKodu], "eth_call", [
```
Burada `await` yaptığımız için yapılacak bütün işleri bir bir yapıp bitmesini bekliyoruz. Yani bağımsız akış (asynchrony) yakalayamıyoruz. Her fetch 1sn sürüyorsa bu kod `6 * adresler.length` zamanda bitecek.  `adresler.length == 12` ise 72 saniye sürecek.

Bitme süresini 1sn'ye indirebilir miyiz?
Hiç `await` komutu kullanmadan yapabilir miyiz?

mnsaglam commented on liste/liste.js:
```diff
@@ -20,43 +20,43 @@ const NodeURLleri = {
+// Adresleri alıp handleları bulur.
+fetch("https://sorular.kimlikdao.org/adresler")
+  .then((res) => res.json())
+  .then(async (adresler) => {
```
Burada da `await`'e gerek yok

```javascript

fetch("https://sorular.kimlikdao.org/adresler")
  .then((res) => res.json())
  .then(handlelarıBul)
  .then(console.log);
  ```
  yapabiliriz.

Roq4 committed:
Adres sorgulaması ve fetch güncellendi

Adresler bir dizi'de toplandı ve sorgular Promise.all kullanılarak eş zamanlı olarak yapıldı.

mnsaglam commented on liste/liste.js:
```diff
@@ -26,27 +26,34 @@ const NodeURLleri = {
+    sonuçlar[ağKodu] = []; // Her ağ için sonuçları saklayacak dizi
+
+    // Her adrese yapılan sorgular map kullanılarak bir dizi Promise oluşturuldu.
+    const sorgular = adresler.map((adres) =>
```
Bunun veri tipi zaten `!Array<!Promise<string>>` değil mi?

Bir alt grupta (46-52) bunu `!Promise<!Array<string>>`'e çeviriyorsun. Promise.all tam olarak bunu yapıyor.

```javascript
Promise.all(adresler.map((adres) => jsonrpc.call())).then((arr) => sonuçlar[ağKodu] = arr)
```

yapsak çok daha kolay değil mi?

Veya `await` istiyorsan (tercih etmiyoruz)

```javascript
sonuçlar[ağKodu] = await Promise.all(adresler.map((adres) => jsonrpc.call()));
```
